apps/opik-backend/src/main/java/com/comet/opik/domain/SpanDAO.java

```diff
@@ -988,14 +997,22 @@ private void bindUpdateParams(SpanUpdate spanUpdate, Statement statement) {
         Optional.ofNullable(spanUpdate.errorInfo())
                 .ifPresent(errorInfo -> statement.bind("error_info", JsonUtils.readTree(errorInfo).toString()));
 
-        if (StringUtils.isNotBlank(spanUpdate.model()) && Objects.nonNull(spanUpdate.usage())) {
-            statement.bind("total_estimated_cost",
-                    ModelPrice.fromString(spanUpdate.model()).calculateCost(spanUpdate.usage()).toString());
-            statement.bind("total_estimated_cost_version", ESTIMATED_COST_VERSION);
+        if (Objects.nonNull(spanUpdate.totalEstimatedCost())) {
```

Comment:
This is mainly for function streams; the != null here would be more straightforward and concise.